### PR TITLE
 #429 #434

### PR DESCRIPTION
**Pull request by 6za**
Addresses some bug found:
- #429 
- #434 
- Also reverted the restriction to APIs as we don't have secrets anymore on UI(Browser Console)
- Added retrial logic for steps that appears flaky on gitlabInstall

Signed-off-by: 6za <53096417+6za@users.noreply.github.com>

### CODE REVIEW

File: internal/helm/helm.go [hunk 1]
```diff
@@ -9,33 +9,47 @@ import (
...
 		// ! commenting out until a clean execution is necessary // create namespace
-		_, _, err := pkg.ExecShellReturnStrings(config.HelmClientPath, "--kubeconfig", config.KubeConfigPath, "repo", "add", "argo", "https://argoproj.github.io/argo-helm")
-		if err != nil {
-			log.Panicf("error: could not run helm repo add %s", err)
-		}
+		// Refers to: https://github.com/kubefirst/kubefirst/issues/434
+		totalAttempts := 5
+		for i := 0; i < totalAttempts; i++ {
+			log.Printf("Installing Argo-CD, attempt (%d of %d)", i+1, totalAttempts)
+			_, _, err := pkg.ExecShellReturnStrings(config.HelmClientPath, "--kubeconfig", config.KubeConfigPath, "repo", "add", "argo", "https://argoproj.github.io/argo-helm")
```
Comment (6za) [inline internal/helm/helm.go:25]: this loop need to be moved to the callers.. or the logic need to be reviewed.

Comment (6za) [reply internal/helm/helm.go:25]: The returns on error will keel the loop, that was not the intended behavior.

Comment (6za) [reply internal/helm/helm.go:27]: logic updated, it should be fine now.

**Review by pagottoo — APPROVED** [review]
lgtm

**Pull request merged**